Client/App.jsx

```diff
@@ -1,24 +1,65 @@
 import React from 'react';
+import MainImage from './mainImage.jsx'
+import SideBar from './sideBar.jsx';
 
 
 
 class App extends React.Component {
   constructor(props) {
     super(props)
+      this.state = {
+        imageList: [],
+        mainImageId: "5f743973c775129696d110f5"
+      }
   }
 
 
+  changeMainImage(imageID) {
+    this.setState({
+      mainImageId: imageID
+    })
+  }
+
+
+  getImagesForEndpoint() {
+    // debugger;
+    fetch('http://localhost:8080/api/products/standard-fit-hoodied-sweatshirt/baby-blue/carousel')
+    .then(res => res.json())
+    .then((images) => {
+      console.log(images);
+      // debugger;
+      this.setState({
+        imageList: images,
+        sideBarImages: images.slice(0, 5)
+      })
+
+    })
+    .catch(err => console.log(err))
+
+  }
+
+  componentDidMount() {
+    this.getImagesForEndpoint()
+  }
 
   render () {
+    return ( this.state.imageList.length > 0 ? <div className="smallCarousel">
```

Comment:
I think adding a comment before this line giving a brief explanation of why you're doing this would help understand it faster.

Reply:
Just added it in thank you